vm/src/api.rs

```diff
@@ -1320,6 +1320,36 @@ macro_rules! record {
     }
 }
 
+#[macro_export]
+macro_rules! hlist_p {
```

Comment:
Maybe check the generated docs and see if it looks fine - macro scoping is weird...